functions/transformations/sum.go

```diff
@@ -109,8 +109,16 @@ type SumIntAgg struct {
 
 func (a *SumIntAgg) DoInt(vs *array.Int64) {
 	// https://issues.apache.org/jira/browse/ARROW-4081
-	if vs.Len() > 0 {
-		a.sum += math.Int64.Sum(vs)
+	if l := vs.Len() - vs.NullN(); l > 0 {
```

Comment:
in a separate PR, I'll check for a table that is empty/all-null and skip this function call completely.